modules/tribes/client/components/TribesPage.component.js

```diff
@@ -0,0 +1,57 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import TribesHeader from './TribesHeader';
+import TribesList from './TribesList';
+import TribesJoinTrustroots from './TribesJoinTrustroots';
+
+import * as api from '../api/tribes.api';
+
+export default function TribesPage({ user, onDisplayPhoto, onHidePhoto, onMembershipUpdated }) {
+  const [tribes, setTribes] = useState([]);
+
+  const handleMembershipUpdated = data => {
+    // update the tribes in state
+
+    setTribes(tribes => tribes.map(tribe => tribe._id === data.tribe._id ? data.tribe : tribe));
```

Comment:
I found this little hard to understand (a couple of arrow functions and ternary in one line, with lots of `tribe` or `tribes` sounding things going on.). It's ok to leave as is but would there be any way explaining this with comment? 😅

Not a blocker. :-)

Reply:
It was me that suggested to change it in this way, perhaps the clearest would be to extract a `mapReplace` function (which has come up in lodash discussion before, e.g. https://github.com/lodash/lodash/issues/1967). Something like `mapReplace(array, condition, item)`, which would replace the item at `condition` with `item`.

Reply:
There is also https://github.com/kolodny/immutability-helper